crates/jarscanner/src/lib.rs

```diff
@@ -0,0 +1,200 @@
+use crate::errors::FileNameError;
+use serde::{Deserialize, Serialize};
+
+use std::collections::HashMap;
+use std::error::Error;
+
+use std::fs::File;
+use std::io::prelude::*;
+use std::path::PathBuf;
+use zip::read::ZipArchive;
+
+pub mod errors;
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Default, Eq)]
+struct DataBlock {
+    entity_path: String,
+    defs: Vec<String>,
+    refs: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Default, Eq)]
+pub struct TargetDescriptor {
+    label_or_repo_path: String,
+    data_blocks: Vec<DataBlock>,
+}
+
+fn non_anon(file_name: &str) -> bool {
+    !((file_name.contains("/$") && file_name.contains("$/")) || file_name.contains("$$anon"))
+}
+
+fn not_in_meta(file_name: &str) -> bool {
+    !file_name.starts_with("META-INF")
```

Comment:
starts with seems risky here. Don't we want `starts_with("META-INF/")` or something? we don't want to have a false positive on `META-INFO` or something.

Reply:
Ah good question: this is what the current JarScanner does, but I think your suggestion makes sense. Will change.